core/src/main/scala/org/apache/spark/crypto/JceAesCtrCryptoCodec.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.security.{GeneralSecurityException, SecureRandom}
+import javax.crypto.Cipher
+import javax.crypto.spec.{IvParameterSpec, SecretKeySpec}
+
+import com.google.common.base.Preconditions
+
+import org.apache.spark.crypto.CommonConfigurationKeys
+.SPARK_SECURITY_JAVA_SECURE_RANDOM_ALGORITHM_DEFAULT
+import org.apache.spark.crypto.CommonConfigurationKeys
+.SPARK_SECURITY_JAVA_SECURE_RANDOM_ALGORITHM_KEY
+import org.apache.spark.{SparkConf, Logging}
+
+/**
+ * Implement the AES-CTR crypto codec using JCE provider.
+ * @param conf
+ */
+private[spark] class JceAesCtrCryptoCodec(conf: SparkConf) extends AesCtrCryptoCodec with Logging {
+  var provider: String = null
+  var random: SecureRandom = null
```

Comment:
I'd rather this be a `val`.

    val random = {
       // all the code that is in setConf() currently
    }